### PR TITLE
Adding RDS Option Groups

### PR DESCRIPTION
**Pull request by tyler-ball**
Copied from https://github.com/chef/chef-provisioning-aws/pull/119

This needs to be completed by adding testing and making sure the resource/provider fit in with the existing rds resource styles.

### CODE REVIEW

File: lib/chef/resource/aws_rds_option_group.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+require 'chef/resource/aws_resource'
+require 'chef/provisioning/aws_driver'
+
+class Chef::Resource::AwsRdsOptionGroup < Chef::Resource::AwsResource
+  self.resource_name = 'aws_rds_option_group'
```
Comment (tyler-ball) [inline lib/chef/resource/aws_rds_option_group.rb]: This should be updated to use the `aws_sdk_type` (even though the option groups don't have a model)

File: lib/chef/resource/aws_rds_option_group.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  actions :create, :delete, :nothing
+  default_action :create
+
+  attribute :name, :kind_of => String, :name_attribute => true
+  attribute :description, :kind_of => String
+  attribute :engine_name, :kind_of => String
+  attribute :major_engine_version, :kind_of => String
+  attribute :options
+  attribute :apply_immediately, :default => false
+
```
Comment (tyler-ball) [inline lib/chef/resource/aws_rds_option_group.rb]: This needs the `aws_object` method, which can just return a hash for now

File: lib/chef/provider/aws_rds_option_group.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
+require 'chef/resource/aws_resource'
+require 'chef/provider/aws_provider'
+
+class Chef::Provider::AwsRdsOptionGroup < Chef::Provider::AwsProvider
+
```
Comment (tyler-ball) [inline lib/chef/provider/aws_rds_option_group.rb]: This should be modified to use the `create_aws_object`, etc. methods

File: lib/chef/provider/aws_rds_option_group.rb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  action :create do
+    fail "Can't create an option group without a description" if new_resource.description.nil?
+    fail "Can't create an option group without an engine name" if new_resource.engine_name.nil?
+    fail "Can't create an option group without a major engine version" if new_resource.major_engine_version.nil?
```
Comment (randomcamel) [inline lib/chef/provider/aws_rds_option_group.rb]: Why not mark these `required` in the resource?

**Comment by tyler-ball** [comment]
Closing since this is a deprecated repo

**Pull request closed without merging**